src/sysinfo/nvapi_adapter.cpp

```diff
@@ -28,16 +28,47 @@ namespace dxvk {
                 reinterpret_cast<void*>(
                     GetProcAddress(vkModule, "vkGetInstanceProcAddr")));
 
+        auto vkEnumerateDeviceExtensionProperties =
+            reinterpret_cast<PFN_vkEnumerateDeviceExtensionProperties>(
+                reinterpret_cast<void*>(
+                    GetProcAddress(vkModule, "vkEnumerateDeviceExtensionProperties")));
+
         VkInstance vkInstance = VK_NULL_HANDLE;
         VkPhysicalDevice vkDevice = VK_NULL_HANDLE;
         dxgiVkInteropAdapter->GetVulkanHandles(&vkInstance, &vkDevice);
 
-        m_devicePciBusProperties.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_PCI_BUS_INFO_PROPERTIES_EXT;
-        m_devicePciBusProperties.pNext = nullptr;
+        uint32_t extCount = 0;
+        // Grab last of valid extensions for this device
+        if (VK_SUCCESS != vkEnumerateDeviceExtensionProperties(vkDevice,
+                                                               nullptr,
+                                                               &extCount,
+                                                               nullptr))
+            return false;
+
+        std::vector<VkExtensionProperties> extensions(extCount);
+        if (VK_SUCCESS != vkEnumerateDeviceExtensionProperties(vkDevice,
+                                                               nullptr,
+                                                               &extCount,
+                                                               extensions.data()))
+            return false;
 
+        for (VkExtensionProperties &ext : extensions) {
```

Comment:
What about `for (const auto& ext : extensions)` for consistency? I world also prefer to omit the brace here.

Reply:
Fixed.